src/main/java/dev/joopie/jambot/music/command/PlayCommandHandler.java

```diff
@@ -68,6 +74,42 @@ public RestAction<?> handle(final CommandInteraction event) {
         try {
             final var input = inputOption.getAsString();
 
+            if (input.contains(SPOTIFY_URL)) {
+                if (input.contains("track")) {
+                    String trackId = input.replaceAll("\\?.*", "").substring(input.lastIndexOf("/") + 1);
+                    String spotifyresult = apiSpotifyService.getTrack(trackId);
+
+                    if (spotifyresult.isEmpty()) {
+                        return event.getHook()
+                                .sendMessage("That's bad! We couldn't found anything with the given Spotify URL. Did you copy the right link?");
+
+                    }
+                    final var result = apiYouTubeService.searchForSong(spotifyresult);
+
+                    if (result.isFound()) {
+                        musicService.play(event.getGuild(), event.getUser(), result.getVideoId());
+                        return event.getHook()
+                                .sendMessageEmbeds(createMessageEmbedOfSearchTrack(event.getGuild().getName(), result));
+                    }
+                } else if (input.contains("playlist")) {
+                    AtomicInteger counter = new AtomicInteger();
+                    String playListId = input.replaceAll("\\?.*", "").substring(input.lastIndexOf("/") + 1);
+                    List<String> spotifyresult = apiSpotifyService.getTracksFromPlaylist(playListId);
+
+                    spotifyresult.forEach(result -> {
+                        var playlistresult = apiYouTubeService.searchForSong(result);
+
+                        if (playlistresult.isFound()) {
+                            musicService.play(event.getGuild(), event.getUser(), playlistresult.getVideoId());
+                            counter.getAndIncrement();
```

Comment:
At the moment, no atomic operation is needed. The `forEach` Stream-method is synchronous.

Reply:
This is needed for the counter to display how many tracks are queued. Because that one is referenced in the stream()